### PR TITLE
Addition of BoW algorithm Code

### PR DESCRIPTION
**Pull request by Arushi-N4**
Closes: #24   

- **Title** : Addition of Bag-of-Words Algorithm Code
- **Name:** Arushi Nirala
- **Idenitfy yourself:** SSOC 2024 Contributor

### Describe the add-ons or changes you've made  📃

After learning from online resources, I added code of bag of words (BoW) Algorithm which include

1. Preprocessing: The preprocess_text function converts text to lowercase and removes non-alphabetic characters.
2. Bag-of-Words Vectorization: Using CountVectorizer from scikit-learn, it creates a matrix where each row represents a document and each column represents a word in the vocabulary.
3. Stop Words Removal: Stop words like "the", "is", "are", and "and" are removed during vectorization using stop_words='english'.
4. Feature Extraction: vectorizer.get_feature_names_out() retrieves the feature names (words) after preprocessing and stop word removal.
5. DataFrame Representation: The resulting bag-of-words matrix is converted into a Pandas DataFrame (bow_df), where rows are labeled by document numbers and columns by word names.

### Type of change ☑️
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested? ⚙️
 I tested the code in Jupyter Notebook and VS Code by running it with sample text data, verifying preprocessing steps, and inspecting the generated bag-of-words dataframe for correctness 

### Checklist: ☑️

- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/AI-Code/blob/main/README.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/AI-Code/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Screenshots 📷
image
 

### Note to reviewers 📄
Please review my changes and let me know if any further adjustments or changes  are needed

### CODE REVIEW

File: NLP/Algorithms/Bag of Words Algorithm/BOW.ipynb
[the remark this answers is not in the document]
Comment (Arushi-N4) [reply NLP/Algorithms/Bag of Words Algorithm/BOW.ipynb:1]: Thank you for the feedback. I've noted the requested changes and will proceed with implementing them promptly. I'll keep you updated on my progress.

However, I have a query. Should I add explanation under 'Documentation' folder or 'BagOfWords` folder?

[the remark this answers is not in the document]
Comment (Avdhesh-Varshney) [reply NLP/Algorithms/Bag of Words Algorithm/BOW.ipynb:1]: If you want to add the documentation to explain the use of Bag of words, create a folder `BagOfWords` and write your documentation inside your directory.

**Comment by Arushi-N4** [comment]
I changed the folder name and converted .ipynb into .py file as per your request.

##Screenshot
image

Kindly review my changes

**Review by Avdhesh-Varshney — APPROVED** [review]
@Arushi-N4 PR Approved 🎉

**Pull request merged**